drafts/draft-0005.md

```diff
@@ -0,0 +1,188 @@
+Status: Draft
+Created: 2018-10-30
+```
+
+## Simple Summary
+
+Provides a standard way for multisig accounts to designate where transactions should be submitted for coordination of additional signatures.
+
+## Abstract
+
+An account may designate which multisig server to use in a data entry that points to a server with a `stellar.toml` file. This file contains a `MULTISIG_ENDPOINT` field which is an API where multisig transactions should be submitted. Wallets should implement this SEP so they can support multisig accounts without requiring them to run a multisig service themselves.
+
+## Motivation
+
+Currently there are only a few public tools that allow for multisig accounts, and they all do it by implementing their own signature coordinator services to do so. Wallets or other services that do not implement their own backends have no real way to deal with multisig accounts, so most of them just don't at all. By allowing an account to specify where they want their multisig transactions submitted to, it enables those wallets to interop with existing multisig services without relying on vendor-specific SDKs and APIs.
+
+Additionally, there may be use cases where users do not want their multisig transactions broadcasted to a public coordinator, but would still like to use existing wallets to submit them to a private coordinator.
+
+## Specification
+
+### Account Data
+
+A multisig account adds a data entry with the key `multisig.domain` and the value of a hostname used to resolve the `stellar.toml` file of the multisig service.
+
+Example:
+Key: `multisig.domain`
+Value: `stellarguard.me`
+
+## Multisig Server stellar.toml
+
+Multisig coordinators advertise the existence of their service through the `stellar.toml` file. The top-level parameter `MULTISIG_ENDPOINT` should contain a fully-qualified URL of a multisig coordination service where transactions should be submitted.
+
+Example of `stellar.toml`:
+```
+MULTISIG_ENDPOINT="https://stellarguard.me/api/transactions"
+```
+
+### Multisig Server API Endpoints
+
+Multisig Services must expose two REST API endpoints.
+
+#### Submit Transaction Endpoint
+
+This is the endpoint defined in the `stellar.toml` under the `MULTISIG_ENDPOINT` key.
+
+##### Request
+
+### Request
+
+```
+POST MULTISIG_ENDPOINT
+```
+
+ContentType: `application/json`.
+
+Request Body Parameters:
+
+Name | Type | Description
+-----|------|------------
+`uri` | string | A [SEP-0007 style](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0007.md)  `tx` URI for the transaction to sign.
+
+Example:
+
+`POST https://stellarguard.me/api/transactions `
+
+```json
+{
+  "uri": "web+stellar:tx?xdr=AAAAAHFd0%2BHQV5u%2FY%2FfM3%2BVelUr1IWwSFL8CUDIAUudUdD4MAAAAZAADyI8AAAADAAAAAAAAAAAAAAABAAAAAAAAAAEAAAAAcV3T4dBXm79j98zf5V6VSvUhbBIUvwJQMgBS51R0PgwAAAAAAAAAAACYloAAAAAAAAAAAVR0PgwAAABAEw8ODG0iixkbAHg1aJATAnZS2531PhGauuSvFDad2WxHKzIenUNbc7K5mGiSpe5jvqe19OQCbNFuBjqN11jfBw%3D%3D"
+}
+```
+
```

Comment:
This need to be changed into a proper POST request such as:
{
  "xdr": ...,
  "callback": ...
}

Reply:
I don't think it's supposed to be this way: You, as the designer, are supposed to provide the rationale as to why you impose such a strange use of POST request into the protocol - rather than me, as the reviewer, am suppose to motive the fact that POST request should be used as intended.

> there are other benefits of using SEP-0007 URLs that will help interoperability

Such as?

Reply:
I didn't said it's non-standard, I said it's weird. You still said nothing about the benefits.

Reply:
Here are some of the benefits:

1. Access to SEP-0007 libraries (both for multisig services and wallets that choose complex payment requests beyond the basic encoding that I coded up in the email thread). **This is a general benefit you get when building layers of protocols (think HTTP vs. TCP, why do you use HTTP for websites and not use TCP socket programming for everything?)**
2. Take advantage of any future improvements to SEP-0007 that expand the capabilities of defining a unique payment request.
3. SEP-0007 URLs are standardized and can be hashed to provide a **unique ID** with a guarantee that it is the full request and is not missing any information in the hash value. This can be used by the multisig service _if they want to_ and is logically a very good ID to use.
4. Security guarantees with a well-defined protocol of how to sign payment requests and also how to verify them — also easily usable from a shared open-source library.

Reply:
Ok. Well there's still something wrong about this design - maybe you're not willing to admit it. That would be solved by the alternative SEP-0007 design I sent you without loosing any of the benefits you mentioned.

Reply:
I think we're at an impasse here. I don't know how comfortable I feel making this dependent on an overhaul of SEP-0007

Reply:
if we do a rework of SEP-0007 at any point in time then the expectation is that this SEP would also use that new protocol, whatever it ends up being. agreeing with @pselden here that this is not a reason to block this SEP. marking this thread as resolved.